### DISCUSSION TITLE
Fl_Window::size_range() doesn't seem to work on X11

### DISCUSSION THREAD

**Discussion by MoAlyousef** [fltk/fltk#1020] [Q&A]
Hello

I'm posting this as a question since I'm not sure whether it's an actual issue or not. Maybe WSL lacks something (like extended window manager hints, I'm not sure). When enabling wayland (also on WSL), the size_range call works correctly.
I tried with the grid-simple.cxx program in the examples directory, and it seems to ignore the size_range.
I built fltk in hybrid mode with both X11 and wayland support.

When running the app with `FLTK_BACKEND=x11`, I'm able to resize the app to below the indicated size range.

If it's an actual bug, I can open a proper issue.

Thank you

**Comment by Albrecht-S** [comment]
This is very likely a `WSL` bug.

I tested on my Debian 12/Bookworm under Gnome/Wayland:
- `FLTK_BACKEND=wayland`: OK
- `FLTK_BACKEND=x11`: OK
- Cross-compiled for `Windows/MinGW` using `Wine`: FAIL

... where OK means `size_range` is honored, FAIL means not honored as described by @MoAlyousef.

This means that I **can not confirm** that `size_range()` doesn't work on X11.

Note that `Wine` runs as an `X11` application (under `XWayland`) as does `FLTK_BACKEND=x11`.

**Reply by Albrecht-S** [reply]
Note also that size_range() is honored on original Windows systems, no matter if FLTK was compiled with MinGW or VS.

I suggest to close this question (not a bug).

**Reply by MoAlyousef** [reply]
Thanks @Albrecht-S 

I thought so as well. Thanks for confirming.
